tools/nightly.py

```diff
@@ -328,6 +329,42 @@ def create(self, *, remove_if_exists: bool = False) -> Path:
         self.base_python("-m", "venv", str(self.prefix))
         assert self.is_venv(), "Failed to create virtual environment."
         (self.prefix / ".gitignore").write_text("*\n", encoding="utf-8")
+
+        if LINUX:
+            activate_script = self.prefix / "bin" / "activate"
+            st_mode = activate_script.stat().st_mode
+            activate_script.chmod(st_mode | 0o200)
```

Comment:
Hmm, why this is needed?

Reply:
The activation script is created with file mode `r--r--r--` with Python 3.12+ on Linux. For Python 3.11, the mode is `rw-r--r--`.